src/engine/pieces/pawn.js

```diff
@@ -35,6 +35,16 @@ export default class Pawn extends Piece {
                 moves.splice(moves.indexOf(move), 2)
             }
         })
+        let diagonalMoveOne = Square.at(currentSquare.row+rowPropagation, currentSquare.col+1)
+        if (this.positionIsOnBoard(diagonalMoveOne) && board.getPiece(diagonalMoveOne)){
+            moves.push(diagonalMoveOne)
+        }
+        let diagonalMoveTwo = Square.at(currentSquare.row+rowPropagation, currentSquare.col-1)
+        if (this.positionIsOnBoard(diagonalMoveTwo) && board.getPiece(diagonalMoveTwo)){
+            moves.push(diagonalMoveTwo)
+        }
```

Comment:
This is doing almost the same thing twice. There is probably a slightly nicer way to do it. 

I think either extract a method (inside the pawn) to `addDiagonalMoveIfAllowed` that we can simply call twice, or maybe create an array with both of the diagonal moves in it, and use a foreach loop or a filter to keep only the ones we want. ie
```
[move1, move2].forEach(move => {
        if (this.positionIsOnBoard(move) && board.getPiece(move)){
                moves.push(move)
        }
})
```

or 

```
return moves.concat([m1, m2].filter(move => this.positionIsOnBoard(move) && board.getPiece(move)))
```